hbw/app/javascript/packs/hbw/components/form/file_upload.js.jsx

```diff
@@ -61,15 +62,60 @@ modulejs.define('HBWFormFileUpload', ['React'], (React) => {
 
       const hiddenValue = JSON.stringify({ files });
       const cssClass = cx('hbw-file-upload', params.css_class, { hidden });
-      const errorMessage = env.translator('errors.file_list_field_required');
-      const errorMessageCss = cx('alert', 'alert-danger', { hidden: fileListPresent });
+
+      const errMessageParamAbcent = env.translator(
+        'errors.file_list_name_required',
+        { field_name: name }
+      );
+
+      const errMessageParamEmpty = env.translator(
+        'errors.file_list_name_empty',
+        { field_name: name }
+      );
+
+      const errMessageParamNotMatched = env.translator(
+        'errors.file_list_name_not_matched',
+        { file_list_name: params.file_list_name }
+      );
+
+      const errMessageParamNotUnique = env.translator(
+        'errors.file_list_name_not_unique',
+        { file_list_name: params.file_list_name }
+      );
+
+      const errMessageCss = cx(
+        'alert',
+        'alert-danger'
+      );
+
+      const fieldReferToUnuniqueFileListName = () => (
+        fileListNames.filter(el => params.file_list_name === el).length > 1
+      );
+
+      const showErrorMessage = () => {
+        let errMessage = '';
+        let hideErrMessage = false;
+        if (typeof params.file_list_name === 'undefined') {
+          errMessage = errMessageParamAbcent;
+        } else if (params.file_list_name === null || params.file_list_name === '') {
+          errMessage = errMessageParamEmpty;
+        } else if (!fileListNames.includes(params.file_list_name)) {
+          errMessage = errMessageParamNotMatched;
+        } else if (fieldReferToUnuniqueFileListName()) {
+          errMessage = errMessageParamNotUnique;
+        } else {
+          hideErrMessage = true;
+        }
+        return <div className={errMessageCss} hidden={hideErrMessage}>{errMessage}</div>;
+      };
```

Comment:
Create smth like `errorMessage = () => ...` returns error message or null and don't render it at all if it's null, and use const. Can't see any reason to hide it instead of conditional rendering.